### PR TITLE
fix: permission driven context menu for pages

### LINKED ISSUE #18425: [Bug]: When user tries to add the widget in the page which has view access and then click on the meatball icon on the page the menu icon doesn't pop up
### Description
- Page context menu is not permission driven on first load. It only became permission driven only when the user switches to the specific page.

https://www.loom.com/share/1b9cf318925d4634aee7fce0460c5489

### PR DESCRIPTION
**Pull request by berzerkeer**
## Description

Right now after fetching current page, we also fetch all pages to extract page DSL , in the process the response also returns the user permissions for each page, but the problem is we are not storing this permissions in redux store along with the pages array. Due to that we can't determine what to show/hide in the context menu of each pages. So with this PR added a new `Redux Action` UPDATE_PAGE_LIST which updates the pages array with its respective user permissions. 

TL;DR The page context menu is now permission driven. Show/hide options based on permission.

Fixes #18425

Media
https://www.loom.com/share/2842940b03064f27982d949431ee6900

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/src/sagas/PageSagas.tsx [hunk 1]
```diff
@@ -950,6 +950,7 @@ function* fetchPageDSLSaga(pageId: string) {
       return {
         pageId: pageId,
         dsl: extractCurrentDSL(fetchPageResponse),
+        userPermissions: fetchPageResponse.data.userPermissions,
```
Comment (ankitakinger) [inline app/client/src/sagas/PageSagas.tsx:953]: add optional chaining before `userPermissions` here safe side

Comment (berzerkeer) [reply app/client/src/sagas/PageSagas.tsx:953]: Oh sure.

**Review by ankitakinger — APPROVED** [review]
LGTM
